### ISSUE TITLE
Support restricted mode

### ISSUE THREAD

**Issue by Zlitus** [fabiospampinato/vscode-open-in-finder#5]
Do you think it would be an issue to make it compatible with Restricted mode? Would be quite useful and I don't see where it could create security issues. Thank you for your work 👍.

**Comment by fabiospampinato** [comment]
If vscode provides APIs for doing this it could be made safe.

**Comment by Zlitus** [comment]
VSCode provide a “Workspace Trust API”: you can declare your extension to provide complete, partial or no support in untrusted workspaces using the `untrustedWorkspaces` capability in `package.json`.

The following example declares that the extension is supported completely in untrusted workspaces. In this case, the extension is enabled in untrusted workspaces.

```json
{
	"capabilities": {
		"untrustedWorkspaces": {
			"supported": true
		}
	}
}
```

The next example declares that the extension is not supported in untrusted workspaces. In this case, the extension is disabled in untrusted workspaces.

```json
{
	"capabilities": {
		"untrustedWorkspaces": {
			"supported": false
		}
	}
}
```

If you think your extension don't have issues with untrusted workspaces, it would be amazing 👍

**Comment by fabiospampinato** [comment]
Given that it executes an applescript I don't see how it could be considered safe in general, if vscode didn't have problems with that it would be a bug.

**Comment by Zlitus** [comment]
> Given that it executes an applescript I don't see how it could be considered safe in general

I definitely agree with this statement 😅. However, in this case saying "trusted" / "untrusted" is completely declarative, not related to what you do.

I think the question you should ask yourself is: Can my plugin be abused by a untrusted evil workspace?
If you think it cannot be abused, so you can run in untrusted mode.

**Comment by fabiospampinato** [comment]
Oh so like the extension itself has to decide if it can be trusted or not? 🤔  If so then I guess this can be marked as trusted. That doesn't mean much though if it's just something that the extension decides rather than something that vscode can enforce.

**Comment by Zlitus** [comment]
Yes, exactly. And that would allow me to run your extensions even when I work on a untrusted workspace 👍
